clcache.py

```diff
@@ -765,43 +776,107 @@ def expandCommandLine(cmdline):
     return ret
 
 
+class Argument(object):
+    def __init__(self, name):
+        self.name = name
+
+    def __len__(self):
+        return len(self.name)
+
+    def __str__(self):
+        return "/" + self.name
+
+    def __eq__(self, other):
+        return type(self) == type(other) and self.name == other.name
+
+    def __hash__(self):
+        key = (type(self), self.name)
+        return hash(key)
+
+
+# /NAMEparameter (no space, required parameter).
+class ArgumentT1(Argument):
+    pass
+
+
+# /NAME[parameter] (no space, optional parameter)
+class ArgumentT2(Argument):
+    pass
+
+
+# /NAME[ ]parameter (optional space)
+class ArgumentT3(Argument):
+    pass
+
+
+# /NAME parameter (required space)
+class ArgumentT4(Argument):
+    pass
+
+
 class CommandLineAnalyzer(object):
 
     @staticmethod
-    def parseArgumentsAndInputFiles(cmdline):
+    def _getParameterizedArgumentType(cmdLineArgument):
         argumentsWithParameter = {
-            'Ob', 'Gs', 'Fa', 'Fd', 'Fm',
-            'Fp', 'FR', 'doc', 'FA', 'Fe',
-            'Fo', 'Fr', 'AI', 'FI', 'FU',
-            'D', 'U', 'I', 'Zp',
-            'MP', 'Tc', 'V', 'wd', 'wo',
-            'W', 'Yc', 'Yl', 'Tp', 'we',
-            'w1', 'w2', 'w3', 'w4', 'Wv',
-            'Yu', 'Zm', 'F', 'Fi',
+            # /NAMEparameter
+            ArgumentT1('Ob'), ArgumentT1('Yl'), ArgumentT1('Zm'),
+            # /NAME[parameter]
+            ArgumentT2('doc'), ArgumentT2('FA'), ArgumentT2('FR'), ArgumentT2('Fr'),
+            ArgumentT2('Gs'), ArgumentT2('MP'), ArgumentT2('Yc'), ArgumentT2('Yu'),
+            ArgumentT2('Zp'), ArgumentT2('Fa'), ArgumentT2('Fd'), ArgumentT2('Fe'),
+            ArgumentT2('Fi'), ArgumentT2('Fm'), ArgumentT2('Fo'), ArgumentT2('Fp'),
+            ArgumentT2('Wv'),
+            # /NAME[ ]parameter
+            ArgumentT3('AI'), ArgumentT3('D'), ArgumentT3('Tc'), ArgumentT3('Tp'),
+            ArgumentT3('FI'), ArgumentT3('U'), ArgumentT3('I'), ArgumentT3('F'),
+            ArgumentT3('FU'), ArgumentT3('w1'), ArgumentT3('w2'), ArgumentT3('w3'),
+            ArgumentT3('w4'), ArgumentT3('wd'), ArgumentT3('we'), ArgumentT3('wo'),
+            ArgumentT3('V'),
+            # /NAME parameter
```

Comment:
That is true. It was created because `/F` and `/FU` are documented as `/NAME param`. But then I found out those are implemented as `/NAME[ ]param`. That was the point where I moved it to number 4 from somewhere in the middle.

I kind of would like to keep it as reference for now, but would also be okay with removing it.